src/infrastructure/transport/fetch.transport.ts

```diff
@@ -117,16 +126,20 @@ export default class FetchTransport {
    * Check response for errors
    * @param response - Response object
    * @param endpoint - API endpoint used for logging
+   * @param isBlob - expected response type is binary
    * @throws Error
    */
-  private async parseResponse(response: Response, endpoint: string): Promise<JSONValue> {
+  private async parseResponse<IsBlob extends boolean = false>(response: Response, endpoint: string, isBlob?: IsBlob): Promise<MaybeJSONValueOrBlob<IsBlob>> {
```

Comment:
this generic looks redundant, can we leave only `isBlob` param?

Reply:
made some changes, i need generic for return type identification